tests/cli/test_cli.py

```diff
@@ -1890,22 +1851,70 @@ def test_search(ctx):
+            _run_command([cl, 'search', 'uuid=' + uuid1 + ',' + uuid2, 'id=.sort', '-u']),
+        )
+        check_equals(
+            uuid2 + '\n' + uuid1, _run_command([cl, 'search', 'name=' + name, 'id=.sort-', '-u'])
+        )
+        check_equals('2', _run_command([cl, 'search', 'name=' + name, '.count']))
+        size1 = float(_run_command([cl, 'info', '-f', 'data_size', uuid1]))
+        size2 = float(_run_command([cl, 'info', '-f', 'data_size', uuid2]))
+        check_equals(
+            size1 + size2, float(_run_command([cl, 'search', 'name=' + name, 'data_size=.sum']))
+        )
+
+        """Search for floating bundles"""
+        check_equals('', _run_command([cl, 'search', '.floating', '-u']))
+        uuid3 = _run_command([cl, 'upload', test_path('a.txt')])
+        _run_command([cl, 'detach', uuid3], 0)
+        check_equals(uuid3, _run_command([cl, 'search', '.floating', '-u']))
+        _run_command([cl, 'rm', uuid3])  # need to remove since not on main worksheet
+
+        """Search bundles on private worksheets"""
+        wuuid = _run_command([cl, 'work', '-u'])
+        new_wuuid = _run_command([cl, 'new', random_name()])
+        _run_command([cl, 'wperm', new_wuuid, 'public', 'n'])  # make worksheet private
+        _run_command([cl, 'work', new_wuuid])  # switch to worksheet
+        uuid = _run_command([cl, 'upload', test_path('a.txt')])
+        check_equals(uuid, _run_command([cl, 'search', uuid, '-u']))
+        _run_command([cl, 'work', wuuid])
+
+        """Search with groups"""
+        # Empty group
+        check_equals('', _run_command([cl, 'search', '.shared']))
+        group_bname = random_name()
+        group_buuid = _run_command([cl, 'upload', test_path('a.txt'), '-n', group_bname])
+        ctx.collect_bundle(group_buuid)
+        user_id, user_name = current_user()
+        # Create new group
+        group_name = random_name()
+        group_uuid = create_group(ctx, group_name)
+        # Make bundle unavailable to public but available to the group
+        _run_command([cl, 'perm', group_buuid, 'public', 'n'])
+        _run_command([cl, 'perm', group_buuid, group_name, 'r'])
+        check_contains(group_buuid[:8], _run_command([cl, 'search', '.shared']))
+        check_contains(group_buuid[:8], _run_command([cl, 'search', 'group={}'.format(group_uuid)]))
+        check_contains(group_buuid[:8], _run_command([cl, 'search', 'group={}'.format(group_name)]))
+
+    # Test with root user.
+    test_search_helper(ctx)
+
+    # Test with non-root user.
+    if not os.getenv('CODALAB_PROTECTED_MODE'):
+        # This test does not work when protected_mode is True.
+        _, current_user_name = current_user()
+        user_name = 'non_root_user_' + random_name()
+        create_user(ctx, user_name, disk_quota='1000000')
+        switch_user(user_name)
+        new_wuuid = _run_command([cl, 'new', random_name()])
+        _run_command([cl, 'work', new_wuuid])
+        test_search_helper(ctx)
```

Comment:
Call helper function here to actually run the tests. Note we call the same helper for the root and non-root user, so the exact same tests are performed in both cases